### PR TITLE
Feature/redis caller

### PR DESCRIPTION
**Pull request by zhenlei520**
1. Fix Redis set error
2. Caller supports handling Service exceptions

### CODE REVIEW

File: src/Caller/MASA.Utils.Caller.Core/AbstractCallerProvider.cs [hunk 1]
```diff
@@ -59,6 +59,8 @@ public virtual async Task<Stream> GetStreamAsync(string? methodName, Cancellatio
     {
         HttpRequestMessage request = CreateRequest(HttpMethod.Get, methodName);
         HttpResponseMessage content = await SendAsync(request, cancellationToken);
+
+
```
Comment (doddgu) [inline src/Caller/MASA.Utils.Caller.Core/AbstractCallerProvider.cs:63]: Why two lines empty

File: src/Caller/MASA.Utils.Caller.Core/DefaultRequestMessage.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    public async Task<TResponse> ProcessResponseAsync<TResponse>(HttpResponseMessage response,
+        CancellationToken cancellationToken = default)
+    {
+        if (response.IsSuccessStatusCode)
+        {
+            switch (response.StatusCode)
+            {
+                case HttpStatusCode.Accepted:
+                case HttpStatusCode.NoContent:
+                    return default;
```
Comment (doddgu) [inline src/Caller/MASA.Utils.Caller.Core/DefaultRequestMessage.cs:21]: Possible null reference return.

File: src/Caller/MASA.Utils.Caller.DaprClient/DaprCallerProvider.cs [hunk 3]
```diff
@@ -32,6 +37,7 @@ public override Task<TResponse> SendGrpcAsync<TResponse>(string methodName, Canc
     public override Task SendGrpcAsync<TRequest>(string methodName, TRequest request, CancellationToken cancellationToken = default)
         => _daprClient.InvokeMethodGrpcAsync(AppId, methodName, request, cancellationToken);
 
-    public override Task<TResponse> SendGrpcAsync<TRequest, TResponse>(string methodName, TRequest request, CancellationToken cancellationToken = default)
+    public override Task<TResponse> SendGrpcAsync<TRequest, TResponse>(string methodName, TRequest request,
+        CancellationToken cancellationToken = default)
```
Comment (doddgu) [inline src/Caller/MASA.Utils.Caller.DaprClient/DaprCallerProvider.cs:41]: Why CancellationToken a single line

Comment (zhenlei520) [reply src/Caller/MASA.Utils.Caller.DaprClient/DaprCallerProvider.cs:41]: Generated by the compiler formatting code, already processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**